File: apimsdc/MsdcQUsrList.cpp
/**
  * \file MsdcQUsrList.cpp
  * API code for table TblMsdcQUsrList (implementation)
  * \author Alexander Wirthmueller
  * \date created: 12 Sep 2018
  * \date modified: 12 Sep 2018
  */

#ifdef _WIN32
	#include "stdafx.h"
#endif

#include "MsdcQUsrList.h"

/******************************************************************************
 class MsdcQUsrList
 ******************************************************************************/

MsdcQUsrList::MsdcQUsrList(
			const uint jnum
			, const string stubRefMsdcMPerson
			, const string sref
			, const string stubRefMsdcMUsergroup
			, const string srefIxVState
			, const string titIxVState
			, const string srefIxMsdcVLocale
			, const string titIxMsdcVLocale
			, const string srefIxMsdcVUserlevel
			, const string titIxMsdcVUserlevel
		) {
	this->jnum = jnum;
	this->stubRefMsdcMPerson = stubRefMsdcMPerson;
	this->sref = sref;
	this->stubRefMsdcMUsergroup = stubRefMsdcMUsergroup;
	this->srefIxVState = srefIxVState;
	this->titIxVState = titIxVState;
	this->srefIxMsdcVLocale = srefIxMsdcVLocale;
	this->titIxMsdcVLocale = titIxMsdcVLocale;
	this->srefIxMsdcVUserlevel = srefIxMsdcVUserlevel;
	this->titIxMsdcVUserlevel = titIxMsdcVUserlevel;
};

bool MsdcQUsrList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "MsdcQUsrList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubRefMsdcMPerson", "prs", stubRefMsdcMPerson);
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubRefMsdcMUsergroup", "usg", stubRefMsdcMUsergroup);
		extractStringUclc(docctx, basexpath, "srefIxVState", "ste", srefIxVState);
		extractStringUclc(docctx, basexpath, "titIxVState", "ste2", titIxVState);
		extractStringUclc(docctx, basexpath, "srefIxMsdcVLocale", "lcl", srefIxMsdcVLocale);
		extractStringUclc(docctx, basexpath, "titIxMsdcVLocale", "lcl2", titIxMsdcVLocale);
		extractStringUclc(docctx, basexpath, "srefIxMsdcVUserlevel", "ulv", srefIxMsdcVUserlevel);
		extractStringUclc(docctx, basexpath, "titIxMsdcVUserlevel", "ulv2", titIxMsdcVUserlevel);
	};

	return basefound;
};

/******************************************************************************
 class ListMsdcQUsrList
 ******************************************************************************/

ListMsdcQUsrList::ListMsdcQUsrList() {
};

ListMsdcQUsrList::ListMsdcQUsrList(
			const ListMsdcQUsrList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i=0;i<nodes.size();i++) nodes[i] = new MsdcQUsrList(*(src.nodes[i]));
};

ListMsdcQUsrList& ListMsdcQUsrList::operator=(
			const ListMsdcQUsrList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i=0;i<nodes.size();i++) nodes[i] = new MsdcQUsrList(*(src.nodes[i]));

	return *this;
};

ListMsdcQUsrList::~ListMsdcQUsrList() {
	clear();
};

void ListMsdcQUsrList::clear() {
	for (unsigned int i=0;i<nodes.size();i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListMsdcQUsrList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	MsdcQUsrList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListMsdcQUsrList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "MsdcQUsrList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i=0;i<ics.size();i++) {
		rec = new MsdcQUsrList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/MsdcQUsrList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japimsdc/MsdcQUsrList.java
/**
  * \file MsdcQUsrList.h
  * Java API code for record of table TblMsdcQUsrList
  * \author Alexander Wirthmueller
  * \date created: 4 Oct 2018
  * \date modified: 4 Oct 2018
  */

package apimsdc;

import org.w3c.dom.*;
import sbecore.*;

public class MsdcQUsrList {
	
	public MsdcQUsrList(
				int jnum
				, String stubRefMsdcMPerson
				, String sref
				, String stubRefMsdcMUsergroup
				, String srefIxVState
				, String titIxVState
				, String srefIxMsdcVLocale
				, String titIxMsdcVLocale
				, String srefIxMsdcVUserlevel
				, String titIxMsdcVUserlevel
			) {
		this.jnum = jnum;
		this.stubRefMsdcMPerson = stubRefMsdcMPerson;
		this.sref = sref;
		this.stubRefMsdcMUsergroup = stubRefMsdcMUsergroup;
		this.srefIxVState = srefIxVState;
		this.titIxVState = titIxVState;
		this.srefIxMsdcVLocale = srefIxMsdcVLocale;
		this.titIxMsdcVLocale = titIxMsdcVLocale;
		this.srefIxMsdcVUserlevel = srefIxMsdcVUserlevel;
		this.titIxMsdcVUserlevel = titIxMsdcVUserlevel;
	};

	public int jnum;
	public String stubRefMsdcMPerson;
	public String sref;
	public String stubRefMsdcMUsergroup;
	public String srefIxVState;
	public String titIxVState;
	public String srefIxMsdcVLocale;
	public String titIxMsdcVLocale;
	public String srefIxMsdcVUserlevel;
	public String titIxMsdcVUserlevel;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "MsdcQUsrList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubRefMsdcMPerson = Xmlio.extractStringUclc(doc, basexpath, "stubRefMsdcMPerson", "prs", null, 0);
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubRefMsdcMUsergroup = Xmlio.extractStringUclc(doc, basexpath, "stubRefMsdcMUsergroup", "usg", null, 0);
			srefIxVState = Xmlio.extractStringUclc(doc, basexpath, "srefIxVState", "ste", null, 0);
			titIxVState = Xmlio.extractStringUclc(doc, basexpath, "titIxVState", "ste2", null, 0);
			srefIxMsdcVLocale = Xmlio.extractStringUclc(doc, basexpath, "srefIxMsdcVLocale", "lcl", null, 0);
			titIxMsdcVLocale = Xmlio.extractStringUclc(doc, basexpath, "titIxMsdcVLocale", "lcl2", null, 0);
			srefIxMsdcVUserlevel = Xmlio.extractStringUclc(doc, basexpath, "srefIxMsdcVUserlevel", "ulv", null, 0);
			titIxMsdcVUserlevel = Xmlio.extractStringUclc(doc, basexpath, "titIxMsdcVUserlevel", "ulv2", null, 0);
			
			return true;
		};

		return false;
	};
};

